visual-inspection/inspection-backend/inspection/explainer/explainers/new_lime_.py

```diff
@@ -0,0 +1,25 @@
+# new explainer
+import numpy as np
+from skimage.segmentation import felzenszwalb, slic, quickshift, watershed
+from skimage.filters import sobel
+from skimage.color import rgb2gray
+
+
+# method to create segments out of a loaded picture using different methods and settings
+def create_segments(img: np.ndarray, seg_method: str, settings: list) -> np.ndarray:
```

Comment:
Geben die Methoden alle das gleiche "Segmente-Format" zurück? Und ist das für uns passend, oder sollten wir umformatieren?

Reply:
Sie geben alle das gleiche Format zurück: Ein Array in welchem Segmentflächen mit der selben Zahl makiert sind. 
Mit skimage.segmentation.find_boundaries lässt sich das in ein bool-Array umwandeln, indem die Segmentgrenzen als True zu finden sind.

Ob wir das so lassen können weiß ich nicht. Soweit ich es aber verstehe arbeitet Lime  jedoch mit dem gleichen Output, welcher dann "eingefärbt" wird:
in: lime/lime/lime_image.py 
https://github.com/marcotcr/lime/blob/fd7eb2e6f760619c29fca0187c07b82157601b32/lime/lime_image.py#L209

![grafik](https://user-images.githubusercontent.com/95425948/146580535-dfba1bb1-5820-43e7-b21e-6bde18532908.png)

Reply:
OK, prima, damit lässt sich arbeiten. Wir könnten auch überlegen, eine Hilfs-Struktur einzuführen, mit der es einfacher ist, einzelne Segmente "anzusprechen". Aber das können wir immer noch machen, wenn uns der Code zu unübersichtlich wird.